presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetPageSource.java

```diff
@@ -213,6 +214,59 @@ public Page getNextPage()
         }
     }
 
+    private Block buildRowBlock(List<TypeSignatureParameter> parameters, int batchSize, int level)
+    {
+        Block[] blocks = new Block[parameters.size()];
+        List<String> fieldNames = new ArrayList<>();
+        boolean[] nullVector = new boolean[batchSize];
+        for (int i = 0; i < parameters.size(); i++) {
+            NamedTypeSignature namedTypeSignature = parameters.get(i).getNamedTypeSignature();
+            Type fieldType = typeManager.getType(namedTypeSignature.getTypeSignature());
+            fieldNames.add(namedTypeSignature.getName());
+            BlockBuilder blockBuilder = fieldType.createBlockBuilder(new BlockBuilderStatus(), batchSize);
+            for (int j = 0; j < batchSize; j++) {
+                blockBuilder.appendNull();
+            }
+            blocks[i] = blockBuilder.build();
+        }
+
+        List<String> pathFields = new ArrayList<>();
+        List<String[]> paths = requestedSchema.getPaths();
+        for (int i = 0; i < paths.size(); i++) {
+            String[] path = paths.get(i);
+            if (path.length >= level + 1) {
+                String pathField = path[level];
+                if (pathFields.contains(pathField)) {
+                    continue;
+                }
+                pathFields.add(pathField);
+                int index = fieldNames.indexOf(pathField);
+                if (index != -1) {
```

Comment:
is it even possible for `index` to be `-1`? If not, we can just assert that` index != -1`